### PR TITLE
Draft: Support OTEL Semantics(WIP)

### PR DESCRIPTION
**Pull request by rish691**
1. Decouple access of tags from business logic of enricher.
2. Add support for OTEL tags in both enricher and normalizer. 
3. Added a module called `attribute-utils` for all tag resolution logic

### CODE REVIEW

File: attribute-utils/src/main/java/org/hypertrace/attribute/span/SpanTagResolver.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  // todo: can't find any envoy related attributes in OTEL
+  private static final String envoyOperationNameAttr = RawSpanConstants.getValue(Envoy.ENVOY_OPERATION_NAME);
+  private static final String envoyIngressSpanValue = RawSpanConstants.getValue(Envoy.ENVOY_INGRESS_SPAN);
+  private static final String envoyEgressSpanValue = RawSpanConstants.getValue(Envoy.ENVOY_EGRESS_SPAN);
+
+  public static Boolean checkForEntrySpan(Event event) {
+    Map<String, AttributeValue> attributeMap = event.getAttributes().getAttributeMap();
+    // Figure out if event is entry or exit based on other span attributes
+    Boolean isEntry = null;
+    if (attributeMap.containsKey(OTEL_SPAN_KIND)) {
```
Comment (pavolloffay) [inline attribute-utils/src/main/java/org/hypertrace/attribute/span/SpanTagResolver.java:39]: nit: could we simplify this method? There are too many if statements

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java [hunk 2]
```diff
@@ -16,39 +15,34 @@
...
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_NAMESPACE));
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_OPERATION));
-      return Optional.of(entityBuilder.build());
+    Optional<String> backendURI = DbTagResolver.getMongoURI(event);
+
+    if (backendURI.isEmpty()) {
+      return Optional.empty();
+    }
+
+    if (StringUtils.isEmpty(backendURI.get())) {
```
Comment (pavolloffay) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java:35]: should be this if merged with the one above?

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java [hunk 3]
```diff
@@ -16,39 +15,34 @@
...
+      LOGGER.warn("Unable to infer a redis backend from event: {}", event);
+      return Optional.empty();
     }
-    return Optional.empty();
+
+    final Builder entityBuilder = getBackendEntityBuilder(BackendType.MONGO, backendURI.get(), event);
```
Comment (pavolloffay) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java:40]: why final? it is just noise here

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/RedisBackendResolver.java [hunk 4]
```diff
@@ -25,17 +25,20 @@ public RedisBackendResolver(FQNResolver fqnResolver) {
...
-        SpanAttributeUtils.getStringAttribute(event, RawSpanConstants.getValue(Redis.REDIS_CONNECTION));
-    if (StringUtils.isEmpty(backendURI)) {
+
+    if (StringUtils.isEmpty(backendURI.get())) {
       LOGGER.warn("Unable to infer a redis backend from event: {}", event);
       return Optional.empty();
     }
-    final Builder entityBuilder = getBackendEntityBuilder(BackendType.REDIS, backendURI, event);
+
+    final Builder entityBuilder = getBackendEntityBuilder(
```
Comment (pavolloffay) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/RedisBackendResolver.java:38]: nti: again why final?

File: attribute-utils/src/main/java/org/hypertrace/attribute/span/SpanTagResolver.java (hunk 1 above)
Comment (rish691) [reply attribute-utils/src/main/java/org/hypertrace/attribute/span/SpanTagResolver.java:39]: I will simplify it. I am actually breaking this pr into smaller pr. Will address it there.

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java (hunk 2 above)
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java:35]: yes, will fix

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java (hunk 3 above)
Comment (rish691) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java:40]: yes can be removed.

**Comment by kotharironak** [comment]
clsong this PR.

**Pull request closed without merging**